Source/SIMPLib/REST/PipelineListener.cpp

```diff
@@ -0,0 +1,312 @@
+#include "PipelineListener.h"
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+PipelineListener::PipelineListener(QObject* parent)
+: QObject(parent)
+, m_ErrorLog(nullptr)
+, m_WarningLog(nullptr)
+, m_StatusLog(nullptr)
+, m_StandardOutputLog(nullptr)
+{
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+PipelineListener::~PipelineListener()
+{
+  closeFiles();
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::vector<PipelineMessage> PipelineListener::getMessages()
+{
+  return m_Messages;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::vector<PipelineMessage> PipelineListener::getErrorMessages()
+{
+  std::vector<PipelineMessage> errorMessages;
+
+  size_t numMessages = m_Messages.size();
+  for(size_t i = 0; i < numMessages; i++)
+  {
+    if(m_Messages[i].getType() == PipelineMessage::MessageType::Error)
+    {
+      errorMessages.push_back(m_Messages[i]);
+    }
+  }
+
+  return errorMessages;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::vector<PipelineMessage> PipelineListener::getWarningMessages()
+{
+  std::vector<PipelineMessage> warningMessages;
+
+  size_t numMessages = m_Messages.size();
+  for(size_t i = 0; i < numMessages; i++)
+  {
+    if(m_Messages[i].getType() == PipelineMessage::MessageType::Warning)
+    {
+      warningMessages.push_back(m_Messages[i]);
+    }
+  }
+
+  return warningMessages;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::vector<PipelineMessage> PipelineListener::getStatusMessages()
+{
+  std::vector<PipelineMessage> statusMessages;
+
+  size_t numMessages = m_Messages.size();
+  for(size_t i = 0; i < numMessages; i++)
+  {
+    if(m_Messages[i].getType() == PipelineMessage::MessageType::StatusMessage)
+    {
+      statusMessages.push_back(m_Messages[i]);
+    }
+  }
+
+  return statusMessages;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+std::vector<PipelineMessage> PipelineListener::getStandardOutputMessages()
```

Comment:
None of these getters are changing the object in question, so they should be marked as const.  It lets the compiler optimize better as well as declaring intent.  In addition, the range-based for loop comment from above applies to all of these.